src/rust/engine/process_execution/src/remote/streaming.rs

```diff
@@ -592,7 +596,8 @@ impl crate::CommandRunner for StreamingCommandRunner {
 
     // Record the time that we started to process this request, then compute the ultimate
     // deadline for execution of this request.
-    let deadline_duration = timeout.unwrap_or(Duration::from_secs(5 * 60)); // TODO: Make this configurable.
+    let deadline_duration =
+      Duration::from_secs(self.overall_deadline_secs).add(timeout.unwrap_or(Duration::default()));
```

Comment:
If you use the `+` operator directly you can avoid importing `std::ops::Add`:
```
Duration::from_secs(self.overall_deadline_secs) + timeout.unwrap_or(Duration::default())
```

Reply:
Done.